logging/google/cloud/logging/logger.py

```diff
@@ -91,7 +95,8 @@ def batch(self, client=None):
 
     def _make_entry_resource(self, text=None, info=None, message=None,
                              labels=None, insert_id=None, severity=None,
-                             http_request=None, timestamp=None):
+                             http_request=None, timestamp=None,
+                             resource=_GLOBAL_RESOURCE):
```

Comment:
The server doesn't assign a default resource, it's required that it's in either the entry or the batch.